src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionSource_ISuggestedActionsSourceExperimental.cs

```diff
@@ -0,0 +1,123 @@
+            {
+                AssertIsForeground();
+
+                using var state = _state.TryAddReference();
+                if (state is null)
+                    yield break;
+
+                var workspace = state.Target.Workspace;
+                if (workspace == null)
+                    yield break;
+
+                await workspace.Services.GetRequiredService<IWorkspaceStatusService>().WaitUntilFullyLoadedAsync(cancellationToken).ConfigureAwait(true);
+
+                using (Logger.LogBlock(FunctionId.SuggestedActions_GetSuggestedActionsAsync, cancellationToken))
+                {
+                    var document = range.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
+                    if (document == null)
+                        yield break;
+
+                    // Compute and return the high pri set of fixes and refactorings first so the user
+                    // can act on them immediately without waiting on the regular set.
+                    var highPriSet = GetCodeFixesAndRefactoringsAsync(
+                        state, requestedActionCategories, document, range, _ => null, CodeActionProviderPriority.High, cancellationToken);
+                    await foreach (var set in highPriSet)
+                        yield return set;
+
+                    var lowPriSet = GetCodeFixesAndRefactoringsAsync(
+                        state, requestedActionCategories, document, range, _ => null, CodeActionProviderPriority.Normal, cancellationToken);
+                    await foreach (var set in lowPriSet)
+                        yield return set;
+                }
+            }
+
+            private async IAsyncEnumerable<SuggestedActionSet> GetCodeFixesAndRefactoringsAsync(
+                ReferenceCountedDisposable<State> state,
+                ISuggestedActionCategorySet requestedActionCategories,
+                Document document,
+                SnapshotSpan range,
+                Func<string, IDisposable?> addOperationScope,
+                CodeActionProviderPriority priority,
+                [EnumeratorCancellation] CancellationToken cancellationToken)
+            {
+                var workspace = document.Project.Solution.Workspace;
+                var supportsFeatureService = workspace.Services.GetRequiredService<ITextBufferSupportsFeatureService>();
+
+                var selection = TryGetCodeRefactoringSelection(state, range);
+
+                var fixesTask = GetCodeFixesAsync(
+                    state, supportsFeatureService, requestedActionCategories, workspace, document, range,
+                    addOperationScope, priority, isBlocking: false, cancellationToken);
+                var refactoringsTask = GetRefactoringsAsync(
+                    state, supportsFeatureService, requestedActionCategories, workspace, document, selection,
+                    addOperationScope, priority, isBlocking: false, cancellationToken);
```

Comment:
I can't really speak to this much, but any concern that we're potentially running this twice, both for priority = high and priority = normal?